src/Castle.Windsor/MicroKernel/Resolvers/DefaultDependencyResolver.cs

```diff
@@ -301,25 +300,28 @@ private bool HasComponentInValidState(string key, DependencyModel dependency, Cr
 			return IsHandlerInValidState(handler) && handler.IsBeingResolvedInContext(context) == false;
 		}
 
-		private object ResolveCore(CreationContext context, ISubDependencyResolver contextHandlerResolver, ComponentModel model, DependencyModel dependency)
+		private bool TryResolveCore(CreationContext context, ISubDependencyResolver contextHandlerResolver, ComponentModel model, DependencyModel dependency, out object value)
```

Comment:
This appears to have no functional difference, however is this a good change? Are you sure all of the methods that this calls don't throw exceptions?

Reply:
This change is required in order to distinguish between returning a null value as success and returning a null value as failure. The change is not meant to have any implications with regard to exception handling.

Reply:
Would you prefer I return `(bool Success, object Value)` or introduce an `Option` type into the codebase? (`Option.None<object>()` for failure to resolve versus `Option.Some<object>(null)` for successful resolution of the value `null`)

Reply:
> This change is required in order to distinguish between returning a null value as success and returning a null value as failure. It isn't meant to have any implications with regard to exception handling.

Ah yes, that is the important part of your change, missed that. I haven't had a good enough look/think, but do you think this could break the behaviour of the `ISubDependencyResolver` contract because returning null from them now does something different?

> Would you prefer I return ...

I don't mind the well established .NET `bool TryXXX` pattern.

Reply:
I thought it wouldn't break because of the `subResolver.CanResolve` check. Are there cases where `subResolver.CanResolve` returns true and `subResolver.Resolve` returns null today? If so, maybe they are also bugs fixed by this PR?

Reply:
If you'd like me to research this, any ideas on likely container configurations would be great. If not, I could return `value is object` and leave a note that this preserves previous behavior but we haven't investigated whether the previous behavior is correct.

Reply:
Probably. Could you check the unit tests for sub resolvers and possibly add an extra one if needed to capture this design decision.